### ISSUE TITLE
How to add custom classes while preserving the classes in original COCO after re-training?

### ISSUE THREAD

**Issue by ztcuthbertlo** [matterport/Mask_RCNN#828]
I would like to add custom classes (like the sample adding Ballon) while retain the original 80 classes COCO already has. For example, after re-training with Ballon dataset with pre-trained COCO model, how can I do object detection on classes like Balloon, Person, Cat, Car?

**Comment by CptK1ng** [comment]
You can just adapt your custom config class and write NUM_CLASSES = <number_of_classes> (in your case I think 82). But you need to pay attention, that your dataset has samples for all classes, this means the coco and the balloon classes) After that, you can train the heads.

```
class CustomConfig(Config):
"""Configuration for training on MS COCO.
    Derives from the base Config class and overrides values specific
    to the COCO dataset.
    """
    # Give the configuration a recognizable name
    NAME = "custom"

    # Number of classes (including background)
    NUM_CLASSES =1 + 80 +1  # Background + Coco + Balloon
   
```

**Comment by CptK1ng** [comment]
For only training the heads, 100 to 1000 samples for each class clould be enough. Just try it out.

**Comment by parul19** [comment]
Do we need to get the whole coco dataset (images of all the 80 class)+ images of balloon/new class and then start to train ...or can we use the pretrained weight+ images of balloon/new class and then train.Is this method will work?
Please let me know how to use transfer learning in this..

**Comment by nyck33** [comment]
@parul19 I think once the final layer is modified, you can use the pretrained weights from all other layers and train the modified layers.  This explains it well except the repo link is broken:  [transfer learning](https://medium.com/@c_61011/transfer-learning-with-mask-r-cnn-f50cbbea3d29)

**Comment by alechagonzalo** [comment]
@parul19 how did you do it? I want to add a new class without losing the 80 COCO classes, but I don't know how to do it. Thank you.

**Comment by Syirrus** [comment]
I've been wondering how to do the same thing. I would like to add a few classes to the existing COCO class. I'm using just basic object detection, no Mask_RCNN. How do you do that?

**Comment by ManjuAlchemy** [comment]
need help steps or any documents to achive these,
 
1.How can I retrain the existing ssd_inception_v2_coco_2018_01_28 model with addition class.
my question is 'How to add custom classes while preserving the classes in original'. and
 otherone question is,
2.How can I train a new model only with my own dataset which has only two classes.

**Comment by skabbit** [comment]
@ManjuAlchemy 
I know the answer for second question is described by authors in the article: https://engineering.matterport.com/splash-of-color-instance-segmentation-with-mask-r-cnn-and-tensorflow-7c761e238b46
This quite comprehensive guide to train a model (from scratch or transfer learn the trained) to mask your own classes.
